From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mahesh Patil <maheshp@nvidia.com>
Date: Thu, 3 Aug 2023 12:54:48 -0700
Subject: [PATCH 042/245] NVIDIA: SAUCE: net:phy: add new speeds

Add speed 2500 and 25000 support in phy

Bug 4043836

Change-Id: I44f6a83fd85e2c718d0f1cae7dec9ed279f8252f
Signed-off-by: Mahesh Patil <maheshp@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/linux-stable/+/3070710
Reviewed-by: Narayan Reddy <narayanr@nvidia.com>
Reviewed-by: Ashutosh Jha <ajha@nvidia.com>
GVS: Gerrit_Virtual_Submit <buildbot_gerritrpt@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/linux-stable/+/3108622
Reviewed-by: Satish Seelamsetti <sseelamsetti@nvidia.com>
Tested-by: Satish Seelamsetti <sseelamsetti@nvidia.com>
Reviewed-by: Bitan Biswas <bbiswas@nvidia.com>

diff --git a/drivers/net/phy/swphy.c b/drivers/net/phy/swphy.c
index 5380f952e846e..6a986c8f7ff8d 100644
--- a/drivers/net/phy/swphy.c
+++ b/drivers/net/phy/swphy.c
@@ -8,7 +8,7 @@
  *         Anton Vorontsov <avorontsov@ru.mvista.com>
  *
  * Copyright (c) 2006-2007 MontaVista Software, Inc.
- * Copyright (c) 2023, NVIDIA CORPORATION.  All rights reserved.
+ * Copyright (c) 2023-2024, NVIDIA CORPORATION.  All rights reserved.
  */
 #include <linux/export.h>
 #include <linux/mii.h>
@@ -30,8 +30,10 @@ enum {
 	SWMII_SPEED_10 = 0,
 	SWMII_SPEED_100,
 	SWMII_SPEED_1000,
+	SWMII_SPEED_2500,
 	SWMII_SPEED_5000,
 	SWMII_SPEED_10000,
+	SWMII_SPEED_25000,
 	SWMII_DUPLEX_HALF = 0,
 	SWMII_DUPLEX_FULL,
 };
@@ -54,10 +56,14 @@ static const struct swmii_regs speed[] = {
 		.lpagb = LPA_1000FULL | LPA_1000HALF,
 		.estat = ESTATUS_1000_TFULL | ESTATUS_1000_THALF,
 	},
+	[SWMII_SPEED_2500] = {
+	},
 	[SWMII_SPEED_5000] = {
 	},
 	[SWMII_SPEED_10000] = {
 	},
+	[SWMII_SPEED_25000] = {
+	},
 };
 
 static const struct swmii_regs duplex[] = {
@@ -78,10 +84,14 @@ static const struct swmii_regs duplex[] = {
 static int swphy_decode_speed(int speed)
 {
 	switch (speed) {
+	case 25000:
+		return SWMII_SPEED_25000;
 	case 10000:
 		return SWMII_SPEED_10000;
 	case 5000:
 		return SWMII_SPEED_5000;
+	case 2500:
+		return SWMII_SPEED_2500;
 	case 1000:
 		return SWMII_SPEED_1000;
 	case 100:
-- 
2.43.0

